### PR TITLE
MAINT-28042 : Implement function to retrieve all membershipTypes for a group

### PR DESCRIPTION
**Pull request by ahamdi**
This is an implementation for retrieving all membershipTypes linked to a specified group.

### CODE REVIEW

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java [hunk 1]
```diff
@@ -714,6 +715,53 @@ public Membership findMembership(String id) throws Exception {
         return null;
     }
 
+    @Override
+    public List<MembershipType> findMembershipTypesByGroup (String groupId) throws Exception{
```
Comment (azayati) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java]:
```suggestion
    public List<MembershipType> findMembershipTypesByGroup(String groupId) throws Exception{
```

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java [hunk 2]
```diff
@@ -714,6 +715,53 @@ public Membership findMembership(String id) throws Exception {
...
+        } catch (Exception e) {
+            handleException("Identity operation error: ", e);
+        }
+
+        if (log.isTraceEnabled()) {
+            Tools.logMethodOut(log, LogLevel.TRACE, "findMembershipTypesByGroup", roleTypes);
+        }
+
+        List<MembershipType> membershipTypes = new ArrayList<>();
+        if (roleTypes != null && !roleTypes.isEmpty()) {
```
Comment (azayati) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java:750]: IMO CollectionUtils.isEmpty can be used instead

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java (hunk 2 above)
Comment (ahamdi) [reply component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java:750]: The same code will be used for both cases, I prefer avoiding that to depend less on utilities classes as soon as they are not required.

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**